scripts/mixins/job_special.lua

```diff
@@ -0,0 +1,52 @@
+-- for mobs with job-specific special ability
+
+require("scripts/globals/mixins");
+require("scripts/globals/status");
+
+g_mixins = g_mixins or {};
+
+g_mixins.job_special = function(mob)
+
+    mob:addListener("SPAWN", "JOB_SPECIAL_SPAWN", function(mob)
+        local defaultAbility = {
+            [JOBS.WAR] = jobSpec.MIGHTY_STRIKES,
+            [JOBS.MNK] = jobSpec.HUNDRED_FISTS,
+            [JOBS.WHM] = jobSpec.BENEDICTION,
+            [JOBS.BLM] = jobSpec.MANAFONT,
+            [JOBS.RDM] = jobSpec.CHAINSPELL,
+            [JOBS.THF] = jobSpec.PERFECT_DODGE,
+            [JOBS.PLD] = jobSpec.INVINCIBLE,
+            [JOBS.DRK] = jobSpec.BLOOD_WEAPON,
+            [JOBS.BST] = jobSpec.FAMILIAR,
+            [JOBS.BRD] = jobSpec.SOUL_VOICE,
+            [JOBS.SAM] = jobSpec.MEIKYO_SHISUI,
+            [JOBS.NIN] = jobSpec.MIJIN_GAKURE,
+            [JOBS.DRG] = jobSpec.CALL_WYVERN,
+            [JOBS.SMN] = jobSpec.ASTRAL_FLOW,
+        };
+
+        mob:setLocalVar("usedJobSpec", 0);
+        mob:setLocalVar("useJobSpecAtPct", math.random(40,60));
+        local ability = defaultAbility[mob:getMainJob()];
+        if (mob:getLocalVar("jobSpec") == 0 and ability ~= nil) then
+            mob:setLocalVar("jobSpec", ability);
```

Comment:
instead of setting this here... I think you should just do nothing in spawn except set the percent, and then in JOB_SPECIAL_WS, check the localvar.  if it's zero, then use the ability in the defaults table, otherwise use the ability in the var.  that way, you can override the ability with whatever you want in the script's OnMobSpawn (note: i am PRETTY SURE that OnMobSpawn runs before the listener, but if it doesn't, that can probably be easily fixed)

Reply:
agreed. i made this change below.